File: public/usage-examples/animations/assign_animation_with_script_and_sound-1-example.cpp
#include "splashkit.h"

int main()
{
    animation_script script = load_animation_script("WalkingScript", "kermit.txt");
    animation anim = create_animation();

    // Assign animation and play sound if set to do so
    assign_animation(anim, script, "WalkFront", true);
    write_line("Assigned WalkFront -> script identifier index: %d", animation_index(script, "WalkFront"));

    free_animation(anim);
    free_animation_script(script);
    return 0;
}

File: public/usage-examples/animations/assign_animation_with_script_and_sound-1-example.py
from splashkit import *

script = load_animation_script("WalkingScript", "kermit.txt")
anim = create_animation()

# assign with sound enabled
assign_animation_with_script_and_sound(anim, script, "WalkFront", True)
write_line(f"Assigned WalkFront -> script identifier index: {animation_index(script, 'WalkFront')}")

free_animation(anim)
free_animation_script(script)
